admin/controllers/school.js

```diff
@@ -45,14 +45,22 @@ const getPupils = async (req, res, next) => {
   res.locals.sortOrder = typeof order === 'boolean' ? !order : true
   res.locals.sortClass = order === false ? 'sort up' : 'sort'
   let pupilsFormatted
+  let groupsIndex = []
+
+  try {
+    const groups = await groupService.getGroups(req.user.schoolId)
+    groups.map((obj) => { groupsIndex[obj.id] = obj.name })
```

Comment:
I doubt this line should be in the controller.  The service should do all the work required for the controller.  The service call can make a generic service call then and marshal the result.

Reply:
@jon-shipley getGroups is used in other places, it is only in this case that the mapping is needed. If you wish I can create a new method in the service just to call getGroups and do the mapping? Please confirm.

Reply:
Yes - that sounds like the best approach. 👍

Reply:
@jon-shipley Done.